### PR TITLE
sink: Implement DSG Sink

### PR DESCRIPTION
**Pull request by iamcndi**
### Check List 

Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
 - No code

Code changes

 - Has exported function/method change
 - Has exported variable/fields change
 - Has interface methods change
 - Has persistent data change

Side effects

 - Possible performance regression
 - Increased code complexity
 - Breaking backward compatibility

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation
 - Need to update key monitor metrics in both TiCDC document and official document

### Release note 

```release-note
Please add a release note.
If you don't think this PR needs a release note then fill it with `None`.
```

### CODE REVIEW

File: cdc/sink/tcp_sink.go [hunk 1]
```diff
@@ -0,0 +1,378 @@
...
+	var eventTypeValue int32
+	var schemaName string
+	var tableName string
+
+	if len(rows) == 0 {
+		return nil
+	} else {
+
+		//rowDataList := make([]*vo.BatchRowsInfo,0);
+		rowInfos := make([]*vo.RowInfos, 0);
```
Comment (liuzix) [inline cdc/sink/tcp_sink.go:90]:
```suggestion
		rowInfos := make([]*vo.RowInfos, 0, len(rows));
```

File: cdc/sink/tcp_sink.go [hunk 2]
```diff
@@ -0,0 +1,378 @@
...
+
+		for _, row := range rows {
+			log.Info("show::::::::::::::::::::::::::::: row", zap.Any("row", row))
```
Comment (liuzix) [inline cdc/sink/tcp_sink.go:93]: These logs should be `Debug`.

File: cdc/sink/tcp_sink.go [hunk 3]
```diff
@@ -0,0 +1,378 @@
...
+				eventTypeValue = 3
+			}
+
+			schemaName = row.Table.Schema
+			tableName = row.Table.Table
+
+
+			rowdata := new(vo.RowInfos)
+
+			columnInfos  :=make([]*vo.ColumnVo,0);
```
Comment (liuzix) [inline cdc/sink/tcp_sink.go:115]: The object produced by `make` is not used.

File: cdc/sink/tcp_sink.go [hunk 4]
```diff
@@ -0,0 +1,378 @@
...
+}
+
+func (b *dsgSink) FlushRowChangedEvents(ctx context.Context, resolvedTs uint64) (uint64, error) {
+	log.Info("dsgSocketSink: FlushRowChangedEvents", zap.Uint64("resolvedTs", resolvedTs))
+
+	commitTs,err:=socket.JddmClientFlush(b.sinkURI.Host,resolvedTs)
+	//commitTs,err:=socket.JddmClientFlush("127.0.0.1:9889",resolvedTs)
+	if err != nil {
+		fmt.Println("err=", err) //出错退出
+		return 0, nil
```
Comment (liuzix) [inline cdc/sink/tcp_sink.go:233]:
```suggestion
		return 0, errors.Trace(err)
```

File: cdc/sink/tcp_sink.go [hunk 5]
```diff
@@ -0,0 +1,378 @@
...
+func (b *dsgSink) Initialize(ctx context.Context, tableInfo []*model.SimpleTableInfo) error {
+	return nil
+}
+
+func (b *dsgSink) Close(ctx context.Context) error {
+	return nil
+}
+
+func (b *dsgSink) Barrier(ctx context.Context) error {
+	return nil
```
Comment (liuzix) [inline cdc/sink/tcp_sink.go:357]: Need to flush all data here.

File: cdc/sink/tcp_sink.go [hunk 6]
```diff
@@ -0,0 +1,378 @@
...
+	return columnInfos
+}
+
+// Initialize is no-op for blackhole
+func (b *dsgSink) Initialize(ctx context.Context, tableInfo []*model.SimpleTableInfo) error {
+	return nil
+}
+
+func (b *dsgSink) Close(ctx context.Context) error {
+	return nil
```
Comment (liuzix) [inline cdc/sink/tcp_sink.go:353]: Do we need to release connections?

File: cmd/cdc/configuration.txt [hunk 7]
```diff
@@ -0,0 +1,6 @@
+host=10.0.0.72
```
Comment (liuzix) [inline cmd/cdc/configuration.txt:1]: What configuration is this for?

File: cdc/sink/go_socket.go [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+	"fmt"
+	"log"
+	"net"
+	"os"
+
+)
+
+
+
+//发送信息
```
Comment (ben1009) [inline cdc/sink/go_socket.go:14]: better use english instead

File: cdc/sink/go_socket.go [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+func sender(jsonStr *RowJson) {
+
+	fmt.Println(">>>>>>>>>>>>>>>>>>jsonStr:",jsonStr)
```
Comment (ben1009) [inline cdc/sink/go_socket.go:17]: just used for debug ?

**Review by ben1009 — COMMENTED** [review]
if the pr is not for review, could add [WIP] or [DNM] as the prefiix of the pr title,
 e.g. [WIP] sink(ticdc): Implement DSG Sink

**Comment by hi-rustin** [comment]
PR currently has a large number of conflicts and I will close it for now. Please feel free to reopen it if you want to continue working on it subsequently. Thank you for your contribution!

**Pull request closed without merging**